### ISSUE TITLE
Reverting a change doesn't always appear in the diff

### ISSUE THREAD

**Issue by mkow** [Reviewable/Reviewable#690]
TL;DR: If at some revision all changes to a file were reverted (so that the file is exactly the same as in base) the "act" of undoing the change is hidden from the diff.

Example scenario: (and how it currently behaves on Reviewable)
1. A contributor (C) submits a PR to a project.
2. A maintainer (M) reviews it and leaves one blocking comment, asking to change something.
3. C fixes the issue, but also reverts a part of their patch in another file (either accidentally or thinking that the change become obsolete after the recent fixup).
4. M opens the review again, but sees only the fix for the issue, not the reverted change and approves the whole PR, though they shouldn't.

Another thing which can happen in 3. is when doing an interactive rebase (e.g. after being asked to squash all the fixups and change the commit messages), C may accidentally drop a commit from the PR (this isn't unrealistic if you work with less experienced or sloppy programmers). I, as a reviewer, would like to see that in the review (that's what review is for).

The hidden diffs are available to display after clicking "reveal" next to "+N obsolete" text in the main pane ("Changes"), but this is something which should IMO be shown by default. When I review a PR at some revision, then open it again later and click "show unreviewed diffs" I expect to see the diff between my last review and the current state. But this is not the case, as some changes are hidden from it.

btw. I'm usually using "combine commits" mode, not sure if this behavior is the same in commit-by-commit mode.

**Comment by pkaminski** [comment]
Related issues: #342, #622.   (More later.)

**Comment by pkaminski** [comment]
This is fixed now thanks to @cgiroux86!
